contracts/CourtRoom.sol

```diff
@@ -183,20 +218,41 @@ contract SwearGame is Owned {
 		registeredPlayersCounter--;
 
 	}
-
-
-	function openNewClaim(bytes32 _evidence) public returns (bool) {
+	function openNewClaim(address beneficiary, uint256 blockNumber,
+			uint8 sig_v, bytes32 sig_r, bytes32 sig_s,bytes32 _evidence) public returns (bool) {
 
 		require(registeredPlayers[msg.sender]);
 
-		 claimId  = caseContract.newClaim(msg.sender, _evidence);
+		claimId  = caseContract.newClaim(msg.sender,bytes32(_evidence));
+
+		caseContract.submitPromise(claimId,beneficiary, blockNumber,sig_v, sig_r, sig_s);
 
 		clientsClaimsIds[msg.sender].push(claimId);
 
 		makeJudgement(claimId);
 
 		return true;
+	}
 
+	/// @notice Cash cheque
+	///
+	/// @param beneficiary beneficiary address
+	/// @param blockNumber the promise is valid until this block number
+	/// @param sig_v signature parameter v
+	/// @param sig_r signature parameter r
+	/// @param sig_s signature parameter s
+	/// The digital signature is calculated on the concatenated triplet of contract address,beneficiary and blockNumber
+	function validatePromise(address beneficiary, uint256 blockNumber,
+			uint8 sig_v, bytes32 sig_r, bytes32 sig_s) private returns (bool){
+
+		  //check the exitance of the proof
+      if (beneficiary == 0x0) return false;
```

Comment:
I will remove that.
It is a check that a client indeed submitted a "promise" ..but any how it will be catch 2 lines after that..